opentelemetry-api/tests/metrics/test_instruments.py

```diff
@@ -0,0 +1,748 @@
+        )
+        self.assertIn(
+            "name", create_observable_counter_signature.parameters.keys()
+        )
+        self.assertIs(
+            create_observable_counter_signature.parameters["name"].default,
+            Signature.empty,
+        )
+        create_observable_counter_signature = signature(
+            Meter.create_observable_counter
+        )
+        self.assertIn(
+            "callback", create_observable_counter_signature.parameters.keys()
+        )
+        self.assertIs(
+            create_observable_counter_signature.parameters["callback"].default,
+            Signature.empty,
+        )
+        create_observable_counter_signature = signature(
+            Meter.create_observable_counter
+        )
+        self.assertIn(
+            "unit", create_observable_counter_signature.parameters.keys()
+        )
+        self.assertIs(
+            create_observable_counter_signature.parameters["unit"].default, ""
+        )
+
+        create_observable_counter_signature = signature(
+            Meter.create_observable_counter
+        )
+        self.assertIn(
+            "description",
+            create_observable_counter_signature.parameters.keys(),
+        )
+        self.assertIs(
+            create_observable_counter_signature.parameters[
+                "description"
+            ].default,
+            "",
+        )
+
+    def test_observable_counter_callback(self):
+        """
+        Test that the API for creating a asynchronous counter accepts a callback.
+        Test that the callback function reports measurements.
+        Test that there is a way to pass state to the callback.
+        Test that the instrument accepts positive measurements.
+        Test that the instrument does not accept negative measurements.
+        """
+
+        create_observable_counter_signature = signature(
+            Meter.create_observable_counter
+        )
+        self.assertIn(
+            "callback", create_observable_counter_signature.parameters.keys()
+        )
+        self.assertIs(
+            create_observable_counter_signature.parameters["name"].default,
+            Signature.empty,
+        )
+
+        def callback():
+            yield 1
+
+        with self.assertRaises(AssertionError):
+            with self.assertLogs(level=ERROR):
+                observable_counter = DefaultObservableCounter(
+                    "name", callback()
+                )
+
+        with self.assertLogs(level=ERROR):
+            observable_counter.observe()
+
+        def callback():
+            yield ChildMeasurement(1)
+            yield ChildMeasurement(-1)
+
+        observable_counter = DefaultObservableCounter("name", callback())
+
+        with self.assertRaises(AssertionError):
+            with self.assertLogs(level=ERROR):
+                observable_counter.observe()
+
+        with self.assertLogs(level=ERROR):
+            observable_counter.observe()
```

Comment:
Just confirming my understanding here the second call to observe calls line 344 and is the reason the error is logged?

Reply:
Correct :+1: